firebase-dataconnect merge into main branch (#6290)
firebase/firebase-android-sdk: firebase-dataconnect/CHANGELOG.md

# Unreleased

> * [feature] Initial release of the Data Connect SDK (public preview). See
>   https://firebase.google.com/docs/data-connect/android-sdk for instructions on
>   getting started.

Reviewer: ```suggestion
* [feature] Initial release of the Data Connect SDK (public preview). Learn how to
  [get started](https://firebase.google.com/docs/data-connect/android-sdk)
  with the SDK in your app.
```

> * [feature] Add ability to specify SerializersModule when serializing.

Reviewer: ```suggestion
* [feature] Added ability to specify `SerializersModule` when serializing.
```

> * [feature] CallerSdkType added, to enable tracking generated SDK usage.

Reviewer: ```suggestion
* [feature] Added `CallerSdkType`, which enables tracking of the generated SDK usage.
```

# Unreleased

> * [feature] initial beta release.

Author: Public preview wording has been changed, as suggested: https://github.com/firebase/firebase-android-sdk/pull/6290/commits/9d47d2ca88edd670490b51d40c59ba90d867020c
